packages/jtex/src/imports.ts

```diff
@@ -11,10 +11,37 @@ function label(title: string, commands: string[]) {
   return `${titleBlock}${commands.join('\n')}\n`;
 }
 
-export function renderImports(expandedImports?: string | ExpandedImports): string {
+export function createImportCommands(commands: Set<string>, existingPackages?: string[]) {
+  const sorted = [...commands].sort();
+  const existingSet = new Set(existingPackages);
+  const filtered = existingPackages ? sorted.filter((p) => !existingSet.has(p)) : sorted;
+  return filtered.map((c) => `\\usepackage{${c}}`);
+}
+
+export function createMathCommands(plugins: Record<string, string>): string[] {
+  if (!plugins || Object.keys(plugins).length === 0) return [];
+  return Object.entries(plugins).map(([k, v]) => {
+    const numArgs = v.match(/#([1-9])/g)?.length ?? 0;
+    if (numArgs === 0) return `\\newcommand{${k}}{${v}}`;
+    return `\\newcommand{${k}}[${numArgs}]{${v}}`;
+  });
+}
+
+export function renderImports(
+  expandedImports?: string | ExpandedImports,
+  existingPackages?: string[],
+): string {
   if (!expandedImports || typeof expandedImports === 'string') return expandedImports || '';
-  const imports = label('imports', [...new Set(expandedImports.imports)]);
-  const commands = label('math commands', [...new Set(expandedImports.commands)]);
+  const packages = new Set(expandedImports.imports);
+  const math = Object.fromEntries(
+    Object.entries(expandedImports.commands).sort(([[a], [b]]) => {
```

Comment:
Should this sorting be in `createMathCommands` (like sorting is in `createImportCommands` - getting a little bogged down, but I think this is just sorting to make the output nicer?

Reply:
Removed!